core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -170,30 +170,28 @@ abstract class AbstractIndex[K, V](@volatile var file: File, val baseOffset: Lon
    * @return a boolean indicating whether the size of the memory map and the underneath file is changed or not.
    */
   def resize(newSize: Int): Boolean = {
-    inLock(lock) {
-      val roundedNewSize = roundDownToExactMultiple(newSize, entrySize)
+    val roundedNewSize = roundDownToExactMultiple(newSize, entrySize)
+    var origin: MappedByteBuffer = null
 
+    inLock(lock) {
       if (_length == roundedNewSize) {
-        false
-      } else {
-        val raf = new RandomAccessFile(file, "rw")
-        try {
-          val position = mmap.position()
-
-          /* Windows won't let us modify the file length while the file is mmapped :-( */
-          if (OperatingSystem.IS_WINDOWS)
-            safeForceUnmap()
-          raf.setLength(roundedNewSize)
-          _length = roundedNewSize
-          mmap = raf.getChannel().map(FileChannel.MapMode.READ_WRITE, 0, roundedNewSize)
-          _maxEntries = mmap.limit() / entrySize
-          mmap.position(position)
-          true
-        } finally {
-          CoreUtils.swallow(raf.close(), this)
-        }
+        return false
+      }
+      origin = mmap
+      val raf = new RandomAccessFile(file, "rw")
+      try {
+        val position = mmap.position()
+        raf.setLength(roundedNewSize)
```

Comment:
Looks like you removed windows specific code from here. It will cause failure if this runs inside windows as we are changing the size of the file. Any specific reason to omit this?